crates/matrix-sdk-crypto/src/backups/keys/recovery.rs

```diff
@@ -196,11 +196,11 @@ impl RecoveryKey {
     /// https://spec.matrix.org/unstable/client-server-api/#backup-algorithm-mmegolm_backupv1curve25519-aes-sha2
     pub fn decrypt_v1(
         &self,
-        mac: &str,
         ephemeral_key: &str,
+        mac: &str,
```

Comment:
I'm seeing this a bit late, but in this case of situation, using newtypes may help identifying what the different roles for similar-looking arguments are (and they're basically free when marked as `#[repr(transparent)]`). In case you get bored :grin:

Reply:
Oh I certainly want to get to a stronger typed API, this is stringly typed because of `libolm`.